mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala

```diff
@@ -125,6 +125,7 @@ class RFormula(override val uid: String)
           encoderStages += new StringIndexer()
             .setInputCol(term)
             .setOutputCol(indexCol)
+          prefixesToRewrite(indexCol + "_") = term + "_"
```

Comment:
With the bug fix, this now needs to be added.  RFormula tests for factor-numeric and factor-factor interactions fail otherwise because of prefixes like "stridx_ie94k9ef90127_"